draco/analysis/ringmapmaker.py

```diff
@@ -386,7 +387,166 @@ def process(self, hstream):
+
+        beam_width = {"X": sig_chime_X, "Y": sig_chime_Y}
+
+        def A(phi, sigma):
+            """A Gaussian like function on the circle."""
+            return np.exp(-((2 * np.tan(phi / 2)) ** 2) / (2 * sigma ** 2))
+
+        def B(phi, u, sigma):
+            """Azimuthal beam transfer function."""
+            return np.exp(2.0j * np.pi * u * np.sin(phi)) * A(phi, sigma)
+
+        sza = hybrid_vis_m.index_map["el"]
+        # TODO: I think the sign is incorrect here
+        dec = np.arcsin(sza) + np.radians(self.telescope.latitude)
+
+        # Create empty ring map
+        d_ew = hybrid_vis_m.index_map["ew"]
+        n_ew = len(d_ew)
+
+        # Determine the weighting coefficient in the EW direction
+        if self.weight_ew == "uniform":
+            weight_ew = np.ones(n_ew)
+        else:  # self.weight_ew == "natural"
+            weight_ew = n_ew - np.arange(n_ew)
+
+        # Exclude the in cylinder baselines if requested (EW = 0)
+        if self.exclude_intracyl:
+            weight_ew[0] = 0.0
+
+        # Normalise the weights
+        weight_ew = weight_ew / weight_ew.sum()
+
+        # Number of RA samples in the final output
+        nra = 2 * hybrid_vis_m.mmax
+
+        # Create ring map, copying over axes/attrs and add the optional datasets
+        rm = containers.RingMap(
+            beam=1, ra=nra, axes_from=hybrid_vis_m, attrs_from=hybrid_vis_m
+        )
+        rm.weight[:] = 0.0
+
+        # This matrix takes the linear combinations of polarisations required to rotate
+        # from XY, YX basis into reXY and imXY
+        # P = np.array(
+        #     [[1, 0, 0, 0], [0, 0.5, 0.5, 0], [0, -0.5j, 0.5j, 0], [0, 0, 0, 1]]
+        # )
+
+        # Make sure ring map is distributed over frequency
+        hybrid_vis_m.redistribute("freq")
+        rm.redistribute("freq")
+
+        # Dereference datasets
+        hv = hybrid_vis_m.vis[:]
+        hw = hybrid_vis_m.weight[:]
+        rmm = rm.map[:]
+        rmw = rm.weight[:]
+
+        phi_arr = np.radians(rm.ra)[np.newaxis, np.newaxis, :]
+
+        pol_map = hybrid_vis_m.index_map["pol"]
+
+        # Loop over frequencies
+        for lfi, fi in hv.enumerate(3):
+
+            freq = hybrid_vis_m.freq[fi]
+            wv = scipy.constants.c * 1e-6 / freq
+
+            u = d_ew / wv
+            u_dec = u[:, np.newaxis] * np.cos(dec)[np.newaxis, :]
+            u_arr = u_dec[:, :, np.newaxis]
+
+            map_m = np.zeros(
+                (hybrid_vis_m.mmax + 1, len(pol_map), len(dec)), dtype=np.complex128
+            )
+
+            # TODO: the handling of the polarisations in here is probably very messed
+            # up.
+            for pi, (pa, pb) in enumerate(pol_map):
+
+                # Get the effective beamwidth for the polarisation combination
+                sig_a = beam_width[pa](freq, dec)
+                sig_b = beam_width[pb](freq, dec)
+                sig = sig_a * sig_b / (sig_a ** 2 + sig_b ** 2) ** 0.5
```

Comment:
What is the order of operations that you expect here?

Reply:
??